orthanc/orthanc-raw/plugin/pixl.py

```diff
@@ -51,36 +49,6 @@
 logger.warning("Running logging at level {}", logging_level)
 
 
```

Comment:
I wonder if we should try this (and registering the the callback again below) instead of sending directly to anon in `process_message`. Could be that it needs to be stable to be kept?
```suggestion
def OnChange(changeType, level, resourceId):  # noqa: ARG001
    """
    # Taken from:
    # https://book.orthanc-server.com/plugins/python.html#auto-routing-studies
    This routes any stable study to a modality named PIXL-Anon if
    should_auto_route returns true
    """
    if changeType != orthanc.ChangeType.STABLE_STUDY:
        return

    # imaging-api didn't get to updating pixl project name, so incomplete
    if _get_project_name_from_study_id(resourceId) == DICOM_TAG_PROJECT_NAME.PLACEHOLDER_VALUE:
        logger.warning("Study {} has not been set with a pixl project name, not sending" resourceId)
        return

    if should_send_to_anon():
        logger.debug("Sending study: {}", resourceId)
        # Although this can throw, since we have nowhere to report errors
        # back to (e.g. an HTTP client), don't try to handle anything here.
        # The client will have to detect that it hasn't happened and retry.
        orthanc_anon_store_study(resourceId)
```

Reply:
I tried adding this back in but ran into the same errors. Turns out the issue was when `study_uid` is an empty string, the function `Orthanc.query_local` would still query by UID and return all studies. Then all but the most recent study was deleted. It's fixed now by checking that `study_uid` is not empty before using it to query studies